src/main/kotlin/graphics/scenery/SettingsEditor.kt

```diff
@@ -0,0 +1,226 @@
+package graphics.scenery
+
+
+import graphics.scenery.utils.LazyLogger
+import net.miginfocom.swing.MigLayout
+import java.awt.Dimension
+import java.awt.event.KeyEvent
+import java.awt.event.KeyListener
+import java.io.File
+import javax.swing.*
+import javax.swing.event.TableModelEvent
+import javax.swing.filechooser.FileNameExtensionFilter
+import javax.swing.table.DefaultTableModel
+
+/**
+ * @author Konrad Michel <konrad.michel@mailbox.tu-dresden.de>
+ *
+ * A Swing UI, that allows the user to load setting in order to manipulate and add settings.
+ * Allows saving and loading of settings into a .properties file
+ * Allows refreshing of the settings table in order to show changes to settings values that happened during runtime.
+ */
+class SettingsEditor(var settings : Settings, private val mainFrame : JFrame = JFrame("SettingsEditor"), width : Int = 480, height : Int = 500) {
```

Comment:
please put "@JvmOverloads" in line 21 so it is more convinient to use in Java

Reply:
If confused on how to add this annotation here, see `SceneryBase` for an example 👍

Reply:
'This annotation is not applicable to target 'class'' -> Also did not do that in the TransferFunctionEditor?!